### PR TITLE
[2.x] revert SlashSyntax to old styles implicits to facilitate source compatibility with sbt 1.x

### PR DESCRIPTION
**Pull request by jtjeferreira**
related #7737

### CODE REVIEW

File: main/src/main/scala/sbt/internal/BuildUtil.scala [hunk 1]
```diff
@@ -93,7 +93,6 @@ object BuildUtil {
   def baseImports: Seq[String] =
     ("import _root_.scala.xml.{TopScope=>$scope}"
       :: "import _root_.sbt.*"
-      :: "import _root_.sbt.given"
```
Comment (jtjeferreira) [inline main/src/main/scala/sbt/internal/BuildUtil.scala:96]: important change: removing `_root_.sbt.given` from `baseImports`

File: main-settings/src/test/scala/sbt/AppendSpec.scala
Comment (jtjeferreira) [inline main-settings/src/test/scala/sbt/AppendSpec.scala:1]: unrelated question: why are these tests disabled/commented? I will enable them in a separate PR...

File: main-settings/src/main/scala/sbt/SlashSyntax.scala
Comment (jtjeferreira) [inline main-settings/src/main/scala/sbt/SlashSyntax.scala:1]: The important changes are in this file.

File: DEVELOPING.md [hunk 2]
```diff
@@ -186,7 +186,7 @@ You'd need alternative DSL import since you can't rely on sbt package object.
 
 ```scala
 // for slash syntax
-import sbt.SlashSyntax0.given
+import sbt.SlashSyntax0.*
```
Comment (eed3si9n) [inline DEVELOPING.md:189]: See https://www.scala-sbt.org/2.x/docs/en/changes/migrating-from-sbt-1.x.html#changing-buildsbt-dsl-to-scala-3x

```scala
import sbt.SlashSyntax0.{ given, * }
```

should work?

Comment (jtjeferreira) [reply DEVELOPING.md:189]: It works, but it makes migration of plugins more complicated...

The downside is that "old style implicits" are not idiomatic scala3. Also my idea was to move to givens in sbt 2.1, but thinking about it, that would be a big breaking change that would require all plugins to be republished...

I will close this

**Pull request closed without merging**